manager/dispatcher/dispatcher.go

```diff
@@ -216,6 +217,9 @@ func (d *Dispatcher) Run(ctx context.Context) error {
 
 	defer cancel()
 	d.ctx, d.cancel = context.WithCancel(ctx)
+	ctx = d.ctx
```

Comment:
Calling `d.cancel()` will cancel `ctx`, because `Context` is an interface, so both `ctx` and `d.ctx` point to the same underlying object (the assignment does not make a copy).

I'm not sure I understood your question properly though.